### PR TITLE
sync: purge products with a negative id

### PR DESCRIPTION
**Pull request by mssola**
## Description

Purge products with a negative ID if they are on the database.

In order to test it, make sure that you have products with a negative ID (e.g. by calling `sync` before applying this patch). After applying this, you should be able to call `sync` and check that there are no products with a negative ID.

## Change Type

*Please select the correct option.*

- [x] **Bug Fix** (a non-breaking change which fixes an issue)
- [ ] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [x] I have verified that my code follows RMT's coding standards with `rubocop`.
- [x] I have reviewed my own code and believe that it's ready for an external review.
- [x] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [x] RMT's test coverage remains at 100%.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Other Notes

Please use this space to provide notes or thoughts to the team, such as tips on how to review/demo your changes.

### CODE REVIEW

**Comment by mssola** [comment]
[[RDQ] Track non-SUSE products (SUMa clients)](https://trello.com/c/ps9jNT3R/2171-rdq-track-non-suse-products-suma-clients)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/231673/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 5f1433c78438a79f9616957dfb1ede0bd116e1bc into 62e114830c657a2bbfb31937658f417267316861)

File: lib/rmt/scc.rb [hunk 1]
```diff
@@ -23,6 +23,11 @@ def sync
     data.each { |item| create_product(item) }
     data.each { |item| migration_paths(item) }
 
+    # NOTE: bug introduced in SCC where some non-SUSE products with negative IDs
+    # were introduced. This should never have happened since they are managed
+    # elsewhere. Purge them now if this is the case for this registration proxy.
+    Product.where('id < ?', 0).destroy_all
```
Comment (digitaltom) [inline lib/rmt/scc.rb:29]: How about adding this as a migration, so we don't have keep it in our codebase?

Comment (mssola) [reply lib/rmt/scc.rb:29]: Ah, of course!

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/231893/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 860ebf25bccdb70d0093416898c635b3d9a5ba43 into 62e114830c657a2bbfb31937658f417267316861)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/231898/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 860ebf25bccdb70d0093416898c635b3d9a5ba43 into d487fbb1c8558fcc13e137be8793d041c02ee199)

File: package/obs/rmt-server.changes [hunk 2]
```diff
@@ -1,10 +1,15 @@
+-------------------------------------------------------------------
+Tue Jan 11 16:52:16 UTC 2022 - Miquel Sabate Sola <msabate@suse.com>
+
+- Remove products with a negative ID on sync
```
Comment (digitaltom) [inline package/obs/rmt-server.changes:4]:
```suggestion
- Remove products with a negative ID during migration
```

**Review by digitaltom — APPROVED** [review]

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/231920/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 42722c2d81cf57fb0f8428b0fd8e12f5f823823c into d487fbb1c8558fcc13e137be8793d041c02ee199)

**Pull request merged**